napari/utils/events/containers/_list.py

```diff
@@ -205,17 +206,69 @@ def move(self, cur_index: int, new_index: int) -> bool:
         """Insert object at ``cur_index`` before ``new_index``.
 
         Both indices refer to the list prior to any object removal
+        (pre-move space).
         """
         if new_index > cur_index:
             new_index -= 1
 
         self.events.moving(index=cur_index, new_index=new_index)
-        item = self._list.pop(cur_index)
-        self._list.insert(new_index, item)
+        with self.events.blocker():
+            item = self.pop(cur_index)
+            self.insert(new_index, item)
         self.events.moved(index=cur_index, new_index=new_index, value=item)
         self.events.reordered(value=self)
         return True
 
+    def move_multiple(self, sources: Sequence[Index], dest_index: int,) -> int:
+        """Move a batch of indices, to a single destination.
+
+        Note, if the dest_index is higher than any of the source indices, then
+        the resulting position of the moved objects after the move operation
+        is complete will be lower than ``dest_index``.
+
+        Parameters
+        ----------
+        sources : Sequence[int or slice]
+            A sequence of indices
+        dest_index : int
+            The destination index.  All sources will be inserted before this
+            index (in pre-move space)
+
+        Returns
+        -------
+        int
+            The number of successful move operations completed.
+
+        Raises
+        ------
+        TypeError
+            If the destination index is a slice, or any of the source indices
+            are not ``int`` or ``slice``.
+        """
+        if isinstance(dest_index, slice):
+            raise TypeError("Destination index may not be a slice")
+
+        to_move = []
+        for idx in sources:
+            if isinstance(idx, slice):
+                to_move.extend(list(range(*idx.indices(len(self)))))
+            elif isinstance(idx, int):
+                to_move.append(idx)
+            else:
+                raise TypeError("Can only move integer or slice indices")
+
+        dest_index -= len([i for i in to_move if i < dest_index])
+
+        self.events.moving(index=to_move, new_index=dest_index)
+        with self.events.blocker():
```

Comment:
I'm still getting problems here with the blocker not blocking as expected - see this comment https://github.com/sofroniewn/napari/pull/6#discussion_r460472336 and one immediately above it. Does that reproduce for you? I think we might need something more like the `block_all` call https://github.com/napari/napari/blob/master/napari/utils/event.py#L766-L771, or modify this context.

Reply:
will look into it.  it's possible I'm misunderstanding the `self.events.blocker()` and that we may need 
```python
with self.events.removed.blocker():
    with self.events.inserted.blocker():
        ...
```
but I want to look through the event code again

Reply:
I found I needed to add something like https://github.com/napari/napari/pull/1504/files#diff-b3f8eaee1d3d919bf96270862a2cebefR840-R883, but I encourage you to take a look too as I maybe missing something also.

Reply:
yes, you're absolutely right.  I was misunderstanding `events.blocker()`,  `block_all` is definitely what we needed instead, and your `blocker_all` context manager is great.

Reply:
pulled in the `blocker_all()` context manager and added a test.